src/components/AddressFormFields.tsx

```diff
@@ -1,172 +1,149 @@
-import React, { FC } from "react";
-import { Form, Col } from "react-bootstrap";
-
-interface AddressFormFieldsProps {
-  setName: (name: string) => void;
-  setEmail: (email: string) => void;
-  setAddress: (address: string) => void;
-  setAddress2: (address2: string) => void;
-  setCity: (city: string) => void;
-  setState: (state: string) => void;
-  setZip: (zip: string) => void;
-  loading: boolean;
-}
-
-const AddressFormFields: FC<AddressFormFieldsProps> = (
-  props: AddressFormFieldsProps
-) => {
+import React, { useState, useEffect } from "react";
+import { Form } from "react-bootstrap";
+import GooglePlacesAutocomplete from 'react-google-places-autocomplete';
+import Select from 'react-select';
+import {getCountries, getStates} from 'country-state-picker'
+import { getCountryFromName } from 'iso-country-utils';
+
+const AddressFormFields = (props) => {
+  const {googleAddress, setGoogleAddress, address} = props;
+  const [countries, setCountries] = useState([]);
+  const [states, setStates] = useState([]);
+
+  useEffect(() => {
+    if(Object.keys(address).length !== 0 && address.country !== "")
```

Comment:
we have an "isNullOrEmpty" helper function in some util file , if it's helpful for this kinda thing

Reply:
need this in that it's a object so checking it wouldn't be a seemless fit